test/test_check.cc

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright (c) 2018-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <sstream>
+#include <vector>
+
+#include <glog/logging.h>
+#include <gtest/gtest.h>
+
+#include "tc/core/check.h"
+
+// XXX:gtest doesn't define a macro that inspects the contents of the exception
+#define ASSERT_THROW_WHAT(x, y)      \
+  try {                              \
+    (x);                             \
+    ASSERT_TRUE(false);              \
+  } catch (std::runtime_error & e) { \
+    ASSERT_EQ(y, e.what());          \
+  }
+
+TEST(CHECK, Plain) {
+  ASSERT_NO_THROW(TC_CHECK(true));
+  {
+    std::stringstream expected;
+    expected << "Check failed [" << __FILE__ << ':' << __LINE__ + 1 << ']';
+    ASSERT_THROW_WHAT(TC_CHECK(false), expected.str());
+  }
+
+  {
+    std::stringstream expected;
+    expected << "Check failed [" << __FILE__ << ':' << __LINE__ + 2 << ']'
+             << ": 1+1=3";
+    ASSERT_THROW_WHAT(TC_CHECK(false) << "1+1=3", expected.str());
+  }
+}
+
+TEST(CHECK, Vector) {
+  std::stringstream expected;
+  expected << "Check failed [" << __FILE__ << ':' << __LINE__ + 3 << ']'
```

Comment:
This looks fragile and probably compiler dependent.
Why can you assume that the reported line number is the last line of
the macro invocation?  Better make sure `ASSERT_THROW_WHAT` call
is within a single line in these tests.

Reply:
Do we still have format checking enabled on CI? If yes, then it will complain.